src/potato_cluster.erl

```diff
@@ -0,0 +1,233 @@
+-module(potato_cluster).
+
+-export([
+	 start_one_pop_verifier/5,
+	 start_single_server_cluster/5,
+	 start_cluster_from_json/2,
+	 stop_cluster/1,
+	 start_web_cluster/1
+	]).
+
+-include_lib("stdlib/include/assert.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+-include("potato_records.hrl").
+
+json_find(Key, Map) when is_atom(Key) ->
+    maps:find(atom_to_binary(Key, utf8), Map).
+
+json_get(Key, Map) when is_atom(Key) ->
+    maps:get(atom_to_binary(Key, utf8), Map).
+
+json_get_str(Key, Map) when is_atom(Key) ->
+    binary_to_list(maps:get(atom_to_binary(Key, utf8), Map)).
+
+json_put(Key, Value, Map) when is_atom(Key) ->
+    maps:put(atom_to_binary(Key, utf8), Value, Map).
+
+start_one_pop_verifier(VerifierArr, JsonConf, MyIndex, UdpServerId, OnEventFn) ->
+
+    MyAddress = (array:get(MyIndex, VerifierArr))#verifier_public_info.network_data,
+
+    {_NetworkAddress, MyNodeNetId} = MyAddress,
+
+    NetSendFn = fun(DestAddressList, MsgId, Data) -> 
+			gen_server:cast({global, UdpServerId}, {send, DestAddressList, {MyAddress, MsgId, Data} })
+		end,
+
+    ConfPrivateKey = default,
+
+    VerifierOnEventFn = fun(Code, Data) -> OnEventFn(MyNodeNetId, Code, Data) end,
+
+    ConfigData = {MyIndex, NetSendFn, VerifierOnEventFn, ConfPrivateKey},
+
+    gen_server:start_link({global, MyNodeNetId}, pop_verifier, {json_map, JsonConf, ConfigData}, []),
+
+    gen_server:cast({global, UdpServerId}, {add_node, MyNodeNetId, {global, MyNodeNetId}}),
+
+    MyNodeNetId.
+
+start_single_server_cluster(VerifierArr, JsonConf, ServerAddress, UdpServerId, OnEventFn) ->
```

Comment:
now all this needs to do is deploy in separate containers!

Reply:
almost there!!

Reply:
this is what this PR is building up to